### ISSUE TITLE
[mechanics] compilation error for SiconosBulletCollisionManager.cpp and ContactTest.cpp

### ISSUE THREAD

**Issue by xhub** [siconos/siconos#115]
Error in SiconosBulletCollisionManager.cpp:

```
[ 92%] Building CXX object mechanics/CMakeFiles/mechanics.dir/src/collision/bullet/SiconosBulletCollisionManager.cpp.o
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:182:8: error: ‘struct BodyShapeRecord’ has virtual functions and accessible non-virtual destructor [-Werror=non-virtual-dtor]
 struct BodyShapeRecord
        ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:221:14: error: base class ‘struct BodyShapeRecord’ has accessible non-virtual destructor [-Werror=non-virtual-dtor]
 SHAPE_RECORD(BodyBoxRecord, SP::SiconosBox, SP::BTBOXSHAPE);
              ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:210:10: note: in definition of macro ‘SHAPE_RECORD’
   struct X : public BodyShapeRecord, std11::enable_shared_from_this<X> { \
          ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:221:14: error: ‘struct BodyBoxRecord’ has virtual functions and accessible non-virtual destructor [-Werror=non-virtual-dtor]
 SHAPE_RECORD(BodyBoxRecord, SP::SiconosBox, SP::BTBOXSHAPE);
              ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:210:10: note: in definition of macro ‘SHAPE_RECORD’
   struct X : public BodyShapeRecord, std11::enable_shared_from_this<X> { \
          ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:222:14: error: base class ‘struct BodyShapeRecord’ has accessible non-virtual destructor [-Werror=non-virtual-dtor]
 SHAPE_RECORD(BodySphereRecord, SP::SiconosSphere, SP::BTSPHERESHAPE);
              ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:210:10: note: in definition of macro ‘SHAPE_RECORD’
   struct X : public BodyShapeRecord, std11::enable_shared_from_this<X> { \
          ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:222:14: error: ‘struct BodySphereRecord’ has virtual functions and accessible non-virtual destructor [-Werror=non-virtual-dtor]
 SHAPE_RECORD(BodySphereRecord, SP::SiconosSphere, SP::BTSPHERESHAPE);
              ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:210:10: note: in definition of macro ‘SHAPE_RECORD’
   struct X : public BodyShapeRecord, std11::enable_shared_from_this<X> { \
          ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:223:14: error: base class ‘struct BodyShapeRecord’ has accessible non-virtual destructor [-Werror=non-virtual-dtor]
 SHAPE_RECORD(BodyCHRecord, SP::SiconosConvexHull, SP::BTCHSHAPE);
              ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:210:10: note: in definition of macro ‘SHAPE_RECORD’
   struct X : public BodyShapeRecord, std11::enable_shared_from_this<X> { \
          ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:223:14: error: ‘struct BodyCHRecord’ has virtual functions and accessible non-virtual destructor [-Werror=non-virtual-dtor]
 SHAPE_RECORD(BodyCHRecord, SP::SiconosConvexHull, SP::BTCHSHAPE);
              ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:210:10: note: in definition of macro ‘SHAPE_RECORD’
   struct X : public BodyShapeRecord, std11::enable_shared_from_this<X> { \
          ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:224:14: error: base class ‘struct BodyShapeRecord’ has accessible non-virtual destructor [-Werror=non-virtual-dtor]
 SHAPE_RECORD(BodyPlaneRecord, SP::SiconosPlane, SP::BTPLANESHAPE);
              ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:210:10: note: in definition of macro ‘SHAPE_RECORD’
   struct X : public BodyShapeRecord, std11::enable_shared_from_this<X> { \
          ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:224:14: error: ‘struct BodyPlaneRecord’ has virtual functions and accessible non-virtual destructor [-Werror=non-virtual-dtor]
 SHAPE_RECORD(BodyPlaneRecord, SP::SiconosPlane, SP::BTPLANESHAPE);
              ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:210:10: note: in definition of macro ‘SHAPE_RECORD’
   struct X : public BodyShapeRecord, std11::enable_shared_from_this<X> { \
          ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:225:14: error: base class ‘struct BodyShapeRecord’ has accessible non-virtual destructor [-Werror=non-virtual-dtor]
 SHAPE_RECORD(BodyCylinderRecord, SP::SiconosCylinder, SP::BTCYLSHAPE);
              ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:210:10: note: in definition of macro ‘SHAPE_RECORD’
   struct X : public BodyShapeRecord, std11::enable_shared_from_this<X> { \
          ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:225:14: error: ‘struct BodyCylinderRecord’ has virtual functions and accessible non-virtual destructor [-Werror=non-virtual-dtor]
 SHAPE_RECORD(BodyCylinderRecord, SP::SiconosCylinder, SP::BTCYLSHAPE);
              ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:210:10: note: in definition of macro ‘SHAPE_RECORD’
   struct X : public BodyShapeRecord, std11::enable_shared_from_this<X> { \
          ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:226:14: error: base class ‘struct BodyShapeRecord’ has accessible non-virtual destructor [-Werror=non-virtual-dtor]
 SHAPE_RECORD(BodyMeshRecord, SP::SiconosMesh, std11::shared_ptr<btSiconosMeshData>);
              ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:210:10: note: in definition of macro ‘SHAPE_RECORD’
   struct X : public BodyShapeRecord, std11::enable_shared_from_this<X> { \
          ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:226:14: error: ‘struct BodyMeshRecord’ has virtual functions and accessible non-virtual destructor [-Werror=non-virtual-dtor]
 SHAPE_RECORD(BodyMeshRecord, SP::SiconosMesh, std11::shared_ptr<btSiconosMeshData>);
              ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:210:10: note: in definition of macro ‘SHAPE_RECORD’
   struct X : public BodyShapeRecord, std11::enable_shared_from_this<X> { \
          ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:227:14: error: base class ‘struct BodyShapeRecord’ has accessible non-virtual destructor [-Werror=non-virtual-dtor]
 SHAPE_RECORD(BodyHeightRecord, SP::SiconosHeightMap, std11::shared_ptr<btSiconosHeightData>);
              ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:210:10: note: in definition of macro ‘SHAPE_RECORD’
   struct X : public BodyShapeRecord, std11::enable_shared_from_this<X> { \
          ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/SiconosBulletCollisionManager.cpp:227:14: error: ‘struct BodyHeightRecord’ has virtual functions and accessible non-virtual destructor [-Werror=non-virtual-dtor]
 SHAPE_RECORD(BodyHeightRecord, SP::SiconosHeightMap, std11::shared_ptr<btSiconosHeightData>);
              ^
```

Error in ContactTest.cpp:
```
[ 97%] Building CXX object mechanics/src/collision/bullet/test/CMakeFiles/testContact.dir/home/xhub/siconos-tmp/mechanics/src/collision/bullet/test/ContactTest.cpp.o
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/test/ContactTest.cpp: In member function ‘void BounceParams::dump()’:
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/test/ContactTest.cpp:56:52: error: ‘struct SiconosBulletOptions’ has no member named ‘breakingThreshold’
     printf("  breakingThreshold:  %.3g\n", options.breakingThreshold);
                                                    ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/test/ContactTest.cpp: In function ‘BounceResult bounceTest(std::string, std::string, const BounceParams&)’:
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/test/ContactTest.cpp:94:12: warning: unused variable ‘desired_bounce_times’ [-Wunused-variable]
     double desired_bounce_times[6]  = { 2.585, 4.645, 6.290, 7.610, 8.660, 9.505 };
            ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/test/ContactTest.cpp:232:9: warning: unused variable ‘N’ [-Wunused-variable]
     int N = ceil((T - t0) / h); // Number of time steps
         ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/test/ContactTest.cpp:258:9: warning: unused variable ‘new_interaction_total’ [-Wunused-variable]
     int new_interaction_total = 0;
         ^
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/test/ContactTest.cpp: In member function ‘void ContactTest::t4()’:
/home/xhub/siconos-tmp/mechanics/src/collision/bullet/test/ContactTest.cpp:504:11: error: ‘struct SiconosBulletOptions’ has no member named ‘breakingThreshold’
   options.breakingThreshold = 0.4;
           ^
```

**Comment by radarsat1** [comment]
Which compiler?

**Comment by xhub** [comment]
gcc 4.9.4

Here is the cmake summary

```
-- The following OPTIONAL packages have been found:

 * PythonInterp
 * Pthread
 * PathFerris
 * LpSolve
 * Git
 * PkgConfig

-- The following REQUIRED packages have been found:

 * PythonFull
 * BLAS
 * LAPACK
 * MUMPS
 * Umfpack
 * SuperLU_MT
 * FCLIB
 * Boost (required version >= 1.47)
 * Bullet
 * HDF5
 * Numpy
 * SWIG (required version >= 2.0.7)
 * Doxygen
 * GMP
 * CppUnit

-- The following OPTIONAL packages have not been found:

 * MlcpSimplex
 * GamsCApi

-- Generate siconos script ...
-- 
============================================ Summary ============================================
-- siconos version 4.1.0 is now ready for compilation and installation.
-- To proceed run 'make' and 'make install' and optionaly 'make test'.
-- C++ Compiler : /usr/bin/c++
-- C++ flags :  -Wall -Wuninitialized -Wno-unused-local-typedef -Wextra -Wno-unused-parameter -Werror=implicit-function-declaration -Werror=array-bounds -Werror=type-limits -Werror=return-type -Werror=overloaded-virtual -Wformat=2 -Werror=format-security -Werror=non-virtual-dtor -D_NUMERICS_INTERNAL_CXX_ -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE
-- C Compiler : /usr/bin/cc
-- C flags : -D__PTHREAD -Wall -Wuninitialized -Wextra -Wno-unused-parameter -Werror=implicit-function-declaration  -Wno-sign-conversion -Wno-error=sign-conversion -Werror=array-bounds -Werror=type-limits -Werror=missing-prototypes -Werror=return-type -Wstrict-overflow=4 -Werror=strict-aliasing -Werror=trampolines -Werror=int-to-pointer-cast -Werror=pointer-to-int-cast -Wformat=2 -Werror=format-security -std=c99 -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_LARGEFILE_SOURCE
-- Fortran Compiler : /usr/bin/gfortran
-- Fortran flags :  -fPIC -w
-- Compilation mode is : DEBUG
-- Code Sources are in : /home/xhub/siconos-tmp
-- Blas is openblas. 
    Blas libraries : /usr/lib64/libopenblas_threads.so.
    Blas headers : cblas.h in /home/xhub/siconos-tmp/externals/blas_lapack/cblas_header.
-- Lapack is generic.
    Lapack libraries : /usr/lib64/libreflapacke.so;/usr/lib64/libreflapack.so;/usr/lib64/libopenblas_threads.so;/usr/lib64/libreflapack.so;/usr/lib64/libopenblas_threads.so;/usr/lib64/libopenblas_threads.so.
    Lapack headers : lapacke.h in /usr/include.
-- Python interpreter is /usr/bin/python3
-- Python libraries are /usr/lib64/libpython3.5m.so
-- All linking libraries are : dl;/usr/lib64/libopenblas_threads.so;/usr/lib64/libreflapacke.so;/usr/lib64/libreflapack.so;/usr/lib64/libopenblas_threads.so;libpthread.so;/home/xhub/softs/gams_ykim/gmstest/libpath47.so;/usr/lib64/liblpsolve55.so.0.0.0;/usr/lib64/libdmumps.so;/usr/lib64/libmumps_common.so;/usr/lib64/libmetis.so;/usr/lib64/libpord.so;umfpack;amd;suitesparseconfig;/usr/lib64/libsuperlu_mt.so;/install/no-asan/lib64/libfclib.so;/usr/lib64/libBulletDynamics.so;/usr/lib64/libBulletCollision.so;/usr/lib64/libLinearMath.so;/usr/lib64/libBulletSoftBody.so;/usr/lib64/libhdf5.so;/usr/lib64/libsz.so;/usr/lib64/libz.so;/usr/lib64/libdl.so;/usr/lib64/libm.so;m;/usr/lib64/libgmp.so.
-- All include directories are : /home/xhub/siconos-tmp/externals/blas_lapack/cblas_header;/usr/include;/usr/include/lpsolve;/usr/include/openblas;/usr/include/metis;/usr/include/superlu_mt;/install/no-asan/include;/usr/include/bullet;/usr/lib64/python3.5/site-packages/numpy/core/include.
-- The project will be installed in /install/siconos-tmp
-- siconos python package will be installed in /install/siconos-tmp/lib64/python3.5/site-packages
-- To get more information about dependencies, config or else, 
-- check CMakeCache.txt file or re-run cmake with -DPRINT_ENV=ON.
-- =================================================================================================

```

**Comment by radarsat1** [comment]
Hopefully fixed but I can't test with gcc 4.9 easily right now.  Makes me wonder if we shouldn't configure Travis at least to use the oldest "supported" compiler?  Also, does Travis build the tests?

**Comment by xhub** [comment]
Thanks it is fixed. The error also appeared here: http://cdash-bipop.inrialpes.fr/viewBuildError.php?buildid=25198

We just have to reduce the number of build failures and test failures on the CDash. But it is always a good idea to check the dashboard after a commit and see if there are any increase within the red boxes
